Presentation.Web/app/components/it-contract/it-contract-overview.controller.ts

```diff
@@ -938,6 +977,15 @@
 
             return concatRoles;
         }
+
+        private getCriticalityName(id: number): string {
+            const criticalitiesWithSameId = this.criticalityOptions.filter(x => x.Id === id);
```

Comment:
searching every time when you don't know how many times you have to do it = bad performance.. build a dictionary/object map, like we do with enums and use that for lookups